### ISSUE TITLE
False positive in shopify/jest/no-if

### ISSUE THREAD

**Issue by BPScott** [Shopify/eslint-plugin-shopify#311]
One more edge case relating to #263.

It warns when the if is within a function, such as a react component you're defining within a test:

```
it('has a child with contentContext', () => {
    const Child: React.SFC = (_props, context) => {
      return context.withinContentContainer ? <div /> : null;
    };
}
```

As discovered in https://github.com/Shopify/polaris-react/pull/1713

**Comment by BPScott** [comment]
I guess this will also stop https://github.com/Shopify/web/pull/14716 throwing a bunch of warnings when web updates to the newest SK too. (That PR changed ifs to ternarys but that loophole was recently closed in #298).
